File: src/main/java/de/thatsich/intellie/common/module/item/AItem.java
package de.thatsich.intellie.common.module.item;

import de.thatsich.intellie.common.module.ATexture;
import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.item.Item;

public abstract class AItem extends Item implements IItem
{
	private final ATexture itemTexture;

	protected AItem ( ATexture itemTexture )
	{
		super();

		this.itemTexture = itemTexture;
	}

	@Override
	public void registerIcons ( final IIconRegister par1IconRegister )
	{
		final String texture = this.itemTexture.getTexture();
		this.itemIcon = par1IconRegister.registerIcon( texture );
	}
}

File: src/main/scala/de/thatsich/minecraft/intellie/common/module/item/AItem.scala
package de.thatsich.minecraft.intellie.common.module.item

import net.minecraft.item.Item
import de.thatsich.minecraft.intellie.common.module.ATexture
import net.minecraft.client.renderer.texture.IIconRegister

/**
 *
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
abstract class AItem(itemTexture: ATexture)
	extends Item
	with IItem
{
	override def registerIcons(iconRegister: IIconRegister)
	{
		val texture: String = this.itemTexture.toString
		this.itemIcon = iconRegister.registerIcon(texture)
	}
}
